### PR TITLE
Add JFR tests

### PR DESCRIPTION
**Pull request by rkennke**
This adds a few tests that allow automatic JFR testing.

The tests can be invoked by:
mx native-unittest com.oracle.svm.jfrtest --build-args -H:+AllowVMInspection

The tests are not under com.oracle.svm.test because a naked 'mx native-unittest' would pick them up and fail because of missing -H:+AllowVMInspection flag.

The test caught one too-strict assert which I removed.

TestSingleEvent demonstrates how a relatively deep parsing test could be implemented (using a couple of helper classes that are taken from OpenJDK's JFR).
TestRecordingParsable and TestJFRCompiles basically test what the name says.
TestClassEvent is a simple test that exercises an event with a Class field, and ensures that the resulting recording is parsable.

In the future (as soon as we generate complete recordings with events), a test like TestClassEvent can be extended to check for the actual events, using JFR APIs.

### CODE REVIEW

**Comment by jiekang** [comment]
Food for long term thought: We'll have to consider how we want to clean the system between runs to keep results deterministic. Currently it will create one binary that runs the whole junit suite. As more tests are added that produce events, the JFR system will be exercised quite a bit, but as soon as global buffers are filled the output is no longer deterministic and test asserts on the output will become more... fun.

So maybe looking into cleaning things between runs in the code or configuring the suite to build and run multiple binaries.

**Comment by rkennke** [comment]
> Food for long term thought: We'll have to consider how we want to clean the system between runs to keep results deterministic. Currently it will create one binary that runs the whole junit suite. As more tests are added that produce events, the JFR system will be exercised quite a bit, but as soon as global buffers are filled the output is no longer deterministic and test asserts on the output will become more... fun.
> 
> So maybe looking into cleaning things between runs in the code or configuring the suite to build and run multiple binaries.

Right. Internal states of JFR would have te be reset, or multiple binaries used instead.

**Comment by rkennke** [comment]
I just noticed that I'd add some more changes on top of it. Please hold off reviewing this PR for now.

**Review by christianhaeubl — APPROVED** [review]

File: substratevm/src/com.oracle.svm.test/src/com/oracle/svm/jfrtest/TestSingleEvent.java
Comment (jiekang) [inline substratevm/src/com.oracle.svm.test/src/com/oracle/svm/jfrtest/TestSingleEvent.java:88]: @rkennke Is this `p.s.v main` intentional?

Comment (rkennke) [reply substratevm/src/com.oracle.svm.test/src/com/oracle/svm/jfrtest/TestSingleEvent.java:88]: > @rkennke Is this `p.s.v main` intentional?

Kinda. I found it useful to run the test standalone. I can remove it if you prefer that.

**Review by jiekang — APPROVED** [review]
Looks good. Thank you! The verification of metadata/constant pools will be very, very useful. Leave it to your choice on keeping the p.s.v.main

**Pull request merged**
